Add support for multiple algorithms per middleware. (#13)
ovotech/ring-jwt: CHANGELOG.md

# 2.0.0 (2020-08-26)

> IMPORTANT: This is a major release with backward compatibility breaking changes.

## Added

- Support for multiple cryptographic algorithms in a single piece of middleware. The middleware will pull the issuer `iss`

> from the incoming JWT and use it to lookup the appropriate algorithm from the middleware configuration to use for decoding.

Reviewer: Maybe add another sentence here: 

> Note, the `iss` claim is not 'trusted' until verified.

## Changed
- To support the new support for multiple algorithms, the format of configuration has changed so that algorithms are
now keyed by issuer. See the [README](./README.md#usage) for an example.

## Removed
- `issuer` optional algorithm field has been removed. (Issuer check is now implicit based on the lookup of issuer in the

> configuration.) 

Reviewer: I think you're right to remove this (replace `:issuer` with `:issuers` IMO). Supporting different modes of configuration just makes verifying the correctness of this library harder over time, and it's easier for subtle problems in configuration handling to creep in.

Author: Yeah indeed. I'm torn about whether to remove support for passing in a `public-key` object instead of a jwks endpoint too but that can wait for another day.

# 2.0.0 (2020-08-26)

> IMPORTANT: This is a major release with backward compatibility breaking changes.

## Added

> - Support for multiple cryptographic algorithms in a single piece of middleware. The middleware will pull the issuer `iss`

Reviewer: ```suggestion
- Support for multiple issuers (each with their own configuration) in a single piece of middleware. The middleware will pull the issuer `iss`
```

from the incoming JWT and use it to lookup the appropriate algorithm from the middleware configuration to use for decoding.
(Note that the `iss` claim is not "trusted" until signature verification has succeeded.) 

## Changed

> - To support the new support for multiple algorithms, the format of configuration has changed so that algorithms are

Reviewer: ```suggestion
- To support multiple issuers, the format of configuration has changed so that configuration is
```
